json-endpoints/class.wpcom-json-api-update-post-v1-2-endpoint.php

```diff
@@ -616,7 +616,7 @@ function write_post( $path, $blog_id, $post_id ) {
 		if ( ! empty( $media_results['errors'] ) )
 			$return['media_errors'] = $media_results['errors'];
 
-		if ( 'publish' !== $post->post_status && isset( $input['title'] )) {
+		if ( 'publish' !== get_post_status( $post_id ) && isset( $input['title'] )) {
```

Comment:
What do you think about checking for `$new` here instead of going for the status right away? This will allow us to save some processing.